### ISSUE TITLE
Failed to initialize terrascan. error : could not connect to github.com

### ISSUE THREAD

**Issue by SayaliMane05** [accurics/terrascan#1241]
* terrascan version: 1.14.0
* Operating System: Ubuntu

### Description

Running 'terrascan init' or 'terrascan scan' command yields following error -
2022-05-09T07:55:24.311+0300	error	cli/init.go:42	failed to initialize terrascan. error : could not connect to github.com

### What I Did

Downloaded and installed terrascan in a Dockerfile. Then while running scan or running terrascan init in a pipeline job, getting this error - 
2022-05-09T07:55:24.311+0300	error	cli/init.go:42	failed to initialize terrascan. error : could not connect to github.com

**Comment by patilpankaj212** [comment]
Hello @SayaliMane05,
I couldn't reproduce this issue with the latest version. It is possible that GitHub wasn't reachable at the time you performed init.
Could you please try to reproduce again? Let us know if you are still facing this issue.

**Comment by SayaliMane05** [comment]
Hello @patilpankaj212 ,
I tried this multiple times. Still getting the same error. 
What could be the reason for this error? Do we need to install anything (any packages etc) along with Terrascan?

**Comment by patilpankaj212** [comment]
No, we don't need to install anything for `terrascan` to perform the `init` operation. When the `init` command is run, it clones the terrascan github repo to `$USER_HOME/.terrascan` on the users machine. So, when init is run, the machine should be connected to internet and github.com should be reachable.

**Comment by SayaliMane05** [comment]
Okay @patilpankaj212
Thanks for the help. I'll check again why it's failing

**Comment by sydneyscoffield** [comment]
I am getting the same error. I am attempting to use the tenable/terrascan docker image from _hub.docker.com_ in my _gitlab-ci.yml_ file and whenever I run any of the terrascan commands, I get this error: 

![Screen Shot 2022-08-25 at 8 49 55 AM](https://user-images.githubusercontent.com/46692338/186697522-bd6578fb-18d8-485b-8ac3-f91b548ba8ea.png)

This is what my .gitlab-ci.yml file looks like. I got this method from the [documentation](https://runterrascan.io/docs/integrations/cicd/). 
![Screen Shot 2022-08-25 at 8 52 39 AM](https://user-images.githubusercontent.com/46692338/186698184-4a049a4e-674e-48a4-bc03-bfdcf19cf8ef.png)

I am using a GitLab shared runner and its a kubernetes executor. I'm not sure if that's the problem or a bug with the container but I would really like to be able to use this image with my GitLab job.

**Comment by ulah07** [comment]
Hi, I run into this error as well. Same as the issuer @SayaliMane05, I also do "terrascan init" in a Dockerfile and build the image through Gitlab pipeline (gitlab shared runner on kubernetes which is same as @sydneyscoffield).

It's weird that it can successfully execute git clone to the target repo_url directly. However, git clone through terrascan init will failed.
![image](https://user-images.githubusercontent.com/118874621/203478176-ac7c9df3-8cee-4353-9d70-143f468c6c70.png)

**Environment**
- terrascan version: 1.16.0
- Docker build image: docker:stable 
- Dockerfile base image: alpine 3.12

**terrascan config**
![image](https://user-images.githubusercontent.com/118874621/203483661-5bba4c2c-17ac-4953-a1df-465d4ec22ad6.png)

**Comment by kverstr** [comment]
Does anyone know where it tries to connect to on github.com?
For us github.com will only be opened on certain context-paths but terrascan is not logging the specifics.


Kr

**Comment by ana-cdc** [comment]
Commenting to say I am also having this issue.

terrascan version: v1.18.3
OS: MacOS Ventura 13.6

**Comment by mmraz** [comment]
I had this problem because our corporate firewall blocks SSH on port 22.  The git@github.com: urls use SSH and we're required to use https git urls instead.  Added the following configuration step in our CI job before running `terrascan init`

`git config --global url."git@github.com:".insteadOf https://github.com/`

**Comment by hboneta** [comment]
> Commenting to say I am also having this issue.
> 
> terrascan version: v1.18.3 OS: MacOS Ventura 13.6

Still an issue with v1.19.9.
